openeogeotrellis/backend.py

```diff
@@ -2819,6 +2833,28 @@ def cancel_job(self, job_id: str, user_id: str):
                                 f"The storage resources for {application_id} could not be found."
                             )
 
+                if get_backend_config().provide_s3_profiles_and_tokens:
+                    secret_name = f's3-{application_id}'
```

Comment:
Isn't it better to get the secret_name from where it was created, instead of regenerating it here.

if I understand correctly it comes from s3_access_cfg_batch_job.yaml.j2  but that is not obvious and consequently a brittle, hidden coupling

Reply:
I agree, I did an attempt on improving the situation. By having a function to get a secret name. I also stopped liking the secret name as it would make it confusing if in the future there is other config that needs to be passed on to the batch-jobs so made the naming more generic.
 
One thing I struggled with was were to put the function. In the end I put in under integrations/kubernetes because it only applies to kubernetes.